libibverbs/device.c

```diff
@@ -262,6 +267,8 @@ struct ibv_context *__ibv_open_device(struct ibv_device *device)
 	context->cmd_fd = cmd_fd;
 	pthread_mutex_init(&context->mutex, NULL);
 
+	ibverbs_device_hold(device);
+
```

Comment:
I think you should do as I suggested, and hold the fd during the open_device and close it once the device is successfully opened - particularly for the downcall into the driver. 

sysfs access outside the opendevice flow needs to be identified and handled via TS checking, I'm not sure how much of that there is..